---
title: "Metagenomics analyse"
author: "Khadija zbair en Mariana Mousa"
date: "2024-04-07"
output:
  html_document:
    theme: readable
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Species identification and abundance estimation

----------Onderdeel 1 ------------

### FastQC analysis

```{r, echo=T, eval=F}
# update conda to the newest version and answer y on question proceed
echo "y" | conda update -n base conda

```

```{bash, engine.opts='-i', echo=T, eval=T}
# show all conda virtual environments
conda env list

# show yml file content
cat ~/daur2/les6/setup_meta_env.yml

```

```{bash, engine.opts='-i', echo=T, eval=F}
# create conda virtual environment
conda env create --file ~/daur2/les6/setup_meta_env.yml
```

```{bash, engine.opts='-i', echo=T, eval=F}

# activate environment
conda activate meta

# install fastqc and answer yes for procceed question
echo "y" | conda install -c bioconda fastqc

# perform FastQC analysis (Forward reads)
fastqc -o ~/daur2/les6/fastqc/data_1/ /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R1_001.fastq.gz

# perform FastQC analysis (Reverse reads)
fastqc -o ~/daur2/les6/fastqc/data_2 /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R2_001.fastq.gz

# deactivate conda virtual environment
conda deactivate
```

```{r , echo=FALSE, fig.height=3, out.width= '100%', fig.cap="Fig. 1: Quality distribution: Right-side: Foward read, Left-side: reverse read. Quality scores range from 0 to 40 on the Phred scale. All bases for all reads have very high quality (>30), indicating that our reads have an accuracy of >99,9%." }

library(png)
library(grid)
library(gridExtra)
img1 <-  rasterGrob(as.raster(readPNG("R2.png")))
img2 <-  rasterGrob(as.raster(readPNG("R1.png")))
grid.arrange(img1, img2, ncol=2, top=textGrob("Per base quality distribution of forward (right) and reverse (left) reads", gp=gpar(fontsize=10,font=8)))
```

De laatste drie bases op de reverse reading(links) hebben een verminderde kwaliteit en een grotere variabiliteit. De gemiddelde kwaliteit (blauwe lijn) ligt echter nog steeds boven de Phred-score 30, daarom hoeven we deze niet uit onze dataset te verwijderen.

--------Onderdeel 2 ------------
# Identificeer de soorten met kraken2 analyse

```{bash, eval=FALSE, engine.opts='-i'}
# activate environment
conda activate meta

#install kraken2 and answer y to proceed question
echo "y" | conda install -c bioconda kraken2

# kraken2 analyse

kraken2 --db /home/daur2/metagenomics/minikraken2_v2_8GB_201904_UPDATE/ --threads 2 --paired --gzip-compressed --output ~/daur2/les6/mock1.kraken --report ~/daur2/les6/mock1.report --use-names /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R1_001.fastq.gz /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R2_001.fastq.gz

conda deactivate

```

Data: 65915836 sequences (19439.90 Mbp) 
Tijd: 977.463s (4046.1 Kseq/m, 1193.29 Mbp/m).

59541308 sequences classified (90.33%)
6374528 sequences unclassified (9.67%)

# Abundance herbereken per geÃ¯dentificeerde soort met bracken

```{bash, eval=F, engine.opts='-i', echo=T}
# activate environment
conda activate meta

echo "y" | conda install -c bioconda bracken

bracken -d /home/daur2/metagenomics/minikraken2_v2_8GB_201904_UPDATE/ -i ~/daur2/les6/mock1.report -o ~/daur2/les6/mock1.bracken

conda deactivate
```
bracken resultaten:
Number of species in sample: 1705 
Number of species with reads > threshold: 1705 
Number of species with reads < threshold: 0 
Total reads in sample: 65915836
Total reads kept at species level (reads > threshold): 39045574
Total reads discarded (species reads < threshold): 0
Reads distributed: 20495136
Reads not distributed (eg. no species above threshold): 598
Unclassified reads: 6374528

kraken2 analyse en bracken analyse gaven dezelfde resultaten, dezelfde aantal unclassified reads: 6374528


kraken-biom backages install en biom file maken

```{bash, eval=T, engine.opts='-i', echo=F}

# activate environment
conda activate meta

# install kraken-biom and answer y to proceed question
echo "y" | conda install -c bioconda kraken-biom 

kraken-biom ~/daur2/les6/mock1_bracken_species.report --fmt json -o ~/daur2/les6/mock1_bracken_species.biom

conda deactivate
```

--------------Onderdeel 3 ------------


# Data visualiseren
```{r plotting}

# install phyloseq package
BiocManager::install("phyloseq")
```

```{r, eval=F}
# load nessecary packages 
library("phyloseq")
library("ggplot2")
data <- "/home/daur2/metagenomics/backup_biomfiles/mock1_bracken_species.biom"

merged_metagenomes <- import_biom(data)


merged_metagenomes@tax_table@.Data <- substring(merged_metagenomes@tax_table@.Data, 4)
colnames(merged_metagenomes@tax_table@.Data) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# rename column header to informative format
colnames(merged_metagenomes@tax_table@.Data) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# inspect data format
View(merged_metagenomes@tax_table@.Data)

# retrieve the present taxonomic kingdoms from our data
unique(merged_metagenomes@tax_table@.Data[,"Kingdom"])
#[1] "Bacteria"  "Eukaryota" "Archaea"   "Viruses"  
```


````{r, eval=FALSE}
merged_metagenomes_b <- subset_taxa(merged_metagenomes, Kingdom == "Bacteria")
sample_sums(merged_metagenomes_b)
sample_names(merged_metagenomes_b) <- "bacteria"
# subset kingdoms
merged_metagenomes_b <- subset_taxa(merged_metagenomes, Kingdom == "Bacteria")
merged_metagenomes_e <- subset_taxa(merged_metagenomes, Kingdom == "Eukaryota")
merged_metagenomes_a <- subset_taxa(merged_metagenomes, Kingdom == "Archaea")
merged_metagenomes_v <- subset_taxa(merged_metagenomes, Kingdom == "Viruses")

# rename samples
sample_names(merged_metagenomes_b) <- "bacteria"
sample_names(merged_metagenomes_e) <- "eukaryote"
sample_names(merged_metagenomes_a) <- "archea"
sample_names(merged_metagenomes_v) <- "virus"

# count number of reads in total
sample_sums(merged_metagenomes)
# count number of reads per kingdom
c(sample_sums(merged_metagenomes_b), sample_sums(merged_metagenomes_e), sample_sums(merged_metagenomes_a), sample_sums(merged_metagenomes_v))
#bacteria eukaryote    archea     virus 
#35817386     14673       327     15873 
````
````{r, eval=FALSE}
# collect kingdoms seperately in data frames
data_b <- data.frame(Samples = sample_names(merged_metagenomes_b),
                   Reads = sample_sums(merged_metagenomes_b))

data_e <- data.frame(Samples = sample_names(merged_metagenomes_e),
                   Reads = sample_sums(merged_metagenomes_e))

data_a <- data.frame(Samples = sample_names(merged_metagenomes_a),
                   Reads = sample_sums(merged_metagenomes_a))

data_v <- data.frame(Samples = sample_names(merged_metagenomes_v),
                   Reads = sample_sums(merged_metagenomes_v))

# merge kingdom data in to one data frame
data_t <- rbind(data_b, data_e, data_a, data_v)
# plot number of reads per kingdom
library(ggplot2)
ggplot(data = data_t, mapping = aes(x = Samples, y = Reads, fill = Samples )) +
  geom_col() +
  theme_classic() +
  ggtitle("Read count per kingdom") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.x=element_text(angle=45, hjust=1))
````


````{r, eval=F}
# reload the biom data, edit column headers and sample name
library("phyloseq")
merged_metagenomes <- import_biom(data)
merged_metagenomes@tax_table@.Data <- substring(merged_metagenomes@tax_table@.Data, 4)
colnames(merged_metagenomes@tax_table@.Data)<- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
colnames(merged_metagenomes@otu_table) <- c("mock1")

# check empty labels (empty = TRUE)
summary(merged_metagenomes@tax_table@.Data== "")



# transform data frame for plotting
glom <- tax_glom(merged_metagenomes, taxrank = "Species")
mock1_metagenome_species <- psmelt(glom)

# combine genus and species label for scientific naming in plot
mock1_metagenome_species$Species <- as.character(mock1_metagenome_species$Species)
mock1_metagenome_species$Species <- paste(mock1_metagenome_species[,]$Genus,mock1_metagenome_species[,]$Species,  sep=" ", collapse=NULL)
# check label manipulation
#unique(mock1_metagenome_species$Species)



mock1_metagenome_species$Species[mock1_metagenome_species$Abundance < 160000] <- "Species < 160.000 abund."

id_species <- ggplot(data=mock1_metagenome_species, aes(x=Sample, y=Abundance, fill=Species))+ geom_bar(aes(), stat="identity", position="stack")
id_species

# transform read count to percentages
glom <- tax_glom(merged_metagenomes, taxrank = "Species")
mock1_metagenome_species_percent <- psmelt(glom)
# normalize read counts to percentages
mock1_metagenome_species_percent$Abundance <- (mock1_metagenome_species_percent$Abundance*100)/sum(mock1_metagenome_species_percent$Abundance)

# combine genus and species label for scientific naming in plot
mock1_metagenome_species_percent$Species <- as.character(mock1_metagenome_species_percent$Species)
mock1_metagenome_species_percent$Species <- paste(mock1_metagenome_species_percent[,]$Genus,mock1_metagenome_species_percent[,]$Species,  sep=" ", collapse=NULL)
mock1_metagenome_species_percent$Species[mock1_metagenome_species_percent$Abundance < 0.5] <- "Species < 0.5% abund."
#unique(mock1_metagenome_species_percent$Species)


id_species_percent <- ggplot(data=mock1_metagenome_species_percent, aes(x=Sample, y=Abundance, fill=Species))+ geom_bar(aes(), stat="identity", position="stack")

# plot species identification
grid.arrange(id_species, id_species_percent, ncol=2)
````



```{r, eval=F}

# collect composition information


mock1_composition <- as.data.frame(read.csv('~/daur2/les6/HU_waternet_MOCK2_composition.csv', row.names=1, sep = ";"))



mock1_composition$amount <- as.numeric(gsub(",", ".", mock1_composition$amount))
colnames(mock1_composition) <- c( "name","amount", "sample_name","total_volume")



# intersect data bases to collect overlap between composition and `kraken2` results

mock1_and_composition_intersect <- mock1_metagenome_species_percent[mock1_metagenome_species_percent$Species %in% mock1_composition$name,]

# collect species that are in composition and not in `kraken2` results
`%!in%` <- Negate(`%in%`) # allow not in = !in

comp_not_in_mock1 <- mock1_composition[mock1_composition$name %!in% mock1_metagenome_species_percent$Species,]

# generate a list of species overlap between mock1 and composition
unique(mock1_and_composition_intersect$Species)

# generate a list of species that are from composition and not in mock1

unique(comp_not_in_mock1$name)

# load library for melting data frame
library(reshape2)

# collect plotting info and pretify layout format
#colnames(mock1_and_composition_intersect)
mock1_and_comp_plotting_data    <- mock1_and_composition_intersect[,c(10,3)]
colnames(mock1_and_comp_plotting_data) <- c("species", "k_abundance")


mock1_and_comp_plotting_data  <- melt(mock1_and_comp_plotting_data, id.var = "species")
mock1_and_comp_plotting_data$value <- as.numeric(mock1_and_comp_plotting_data$value)


```

```{r, eval=F}
ggplot(mock1_and_comp_plotting_data, aes(x = species, y = value, fill = variable)) + 
  geom_bar(aes(), stat="identity", position="dodge") +
  theme_classic() +
  ylab("Abundance (%)") +
  xlab("") +
  ggtitle("Abundance comparison between Kraken2 results and composition") +
  theme(plot.title = element_text(hjust = 0.5)) +
  ylim(0,25) +
  theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust = 1)) +
  scale_fill_manual(values=c("skyblue", "orangered"))
```

```{r}
library(png)
library(grid)
library(gridExtra)
img3 <-  rasterGrob(as.raster(readPNG("ggplot.png.png")))
```
